src/modules/auth/passport/oidc-strategy.ts

```diff
@@ -58,10 +69,29 @@ passport.use(
 
     try {
       const oidcData = await oidcResponse.json();
+
       const tokenDetails = jwtDecode<AuthStoreToken>(oidcData!.id_token);
-      const oidcRoles = tokenDetails.resource_access
-        ? tokenDetails.resource_access[process.env.KEYCLOAK_CLIENT_ID || ''].roles
-        : [];
+      let oidcRoles;
+
+      switch (process.env.OIDC_PROVIDER) {
+          case 'KEYCLOAK': {
+              oidcRoles = tokenDetails.resource_access
+                  ? tokenDetails.resource_access[process.env.OIDC_CLIENT_ID || ''].roles
+                  : [];
+              break;
+          }
+          case 'ENTRA_ID': {
+              oidcRoles = tokenDetails.roles || [];
+              break;
+          }
+          default: {
+              throw new HttpApiException(
+                  HttpStatusCode.InternalServerError,
+                  'Invalid OIDC provider configured.',
+              );
+          }
```

Comment:
Maybe test this earlier in the flow?

Reply:
Too bad the ways roles are passed is not a standardized way. 

I am not well read intro Entra, but does this mean that you can only use a single Entra instance for a single applications (as opposed to Keycloak which you can configure for many applications, hence the required `CLIENT_ID`). Asking since I want to check that the "general" case is implemented here for Entra.

I suggest moving the check if the environment variables to the start of the function. In case `OIDC_PROVIDER=KEYCLOAK`, you should also check if `OIDC_CLIENT_ID` is set (and otherwise throw an error). Please also add checks for any other environment variables that are required and still missing.

Finally, I do not think I ever checked this, but what does Keycloak/Entra return if the requested user does not have any roles at all? Are `tokenDetails.roles` and `tokenDetails.resource_access[].roles` actually `undefined`, or do they provide an empty array? If the latter is the case, the ternary/double pipe can be omitted.

Reply:
To add to this, if the environment variables are not properly configured; I do not think this should be reflected to the users. A non-transparent Internal Server Error should suffice for the user, and pino should log that the environment variables have been set inproperly.

Reply:
This is the "standardized" way that Entra handles added roles according to: https://learn.microsoft.com/en-us/entra/identity-platform/howto-add-app-roles-in-apps. You can create multiple applications that work in all the same way with different client id's (for example aurora and aurora-test, with different redirect URIs and different client secrets), but also different app roles (AFAIK). If that is what you are asking. 

Indeed like you say, OIDC does not standardize where the roles are in the id token, but at least per identity provider this is consistent.

Reply:
Done!